### PR TITLE
Calculate hand scale and use this instead of palm width

### PR DESCRIPTION
**Pull request by rblenkinsopp**
## Summary

This changes the pinch calculation for OpenXR to no longer be dependent on the palm width being the same as it was in LeapC. This also has the advantage of working better on non-Ultraleap OpenXR hand-tracking.

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

- Closes https://ultrahaptics.atlassian.net/browse/UNITY-1142
- Closes https://ultrahaptics.atlassian.net/browse/OXR-308

### CODE REVIEW

File: Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs [hunk 1]
```diff
@@ -320,13 +321,21 @@ private bool PopulateLeapHandFromOpenXRJoints(HandTracker handTracker, ref Hand
             return true;
         }
 
-        private float CalculatePinchStrength(ref Hand hand, float palmWidth)
+        private static readonly float[] DefaultMetacarpalLengths = { 0, 0.6812f, 0.6460f, 0.5800f, 0.5369f };
```
[the remark this answers is not in the document]
Comment (rblenkinsopp) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: Sorry I should have added a comment; they are more LeapC magic variables, specifically from `src/TrackingInterface/CapsuleHand.cpp` and `CapsuleHand::GetDefaultFinger()` methods.

[the remark this answers is not in the document]
Comment (MattGrayUL) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: Is it possible that these values are off by a factor of 10?

I get 0 for pinch strength _always_ with this fix, but if I use e.g. 0.06812f - I get results that match LeapC

[the remark this answers is not in the document]
Comment (MaxPalmer-UH) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: Yes, they seem off.

[the remark this answers is not in the document]
Comment (MattGrayUL) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: I believe this works as we expect when I change the DefaultMetacarpalLengths to match what I mentioned above

I will push it in a sec after I tidy up my logs

[the remark this answers is not in the document]
Comment (rblenkinsopp) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: Yeah they could well be, I thought I'd accounted for this when I converted the by I might have got an order of magnitude out. I expected them to be cm in LeapC...
```
fingers[0].metacarpusLength = 0.0;
fingers[1].metacarpusLength = 68.12;
fingers[2].metacarpusLength = 64.60;
fingers[3].metacarpusLength = 58.00;
fingers[4].metacarpusLength = 53.69;
```

[the remark this answers is not in the document]
Comment (MaxPalmer-UH) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: LeapC units are mm

[the remark this answers is not in the document]
Comment (MaxPalmer-UH) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: https://developer.leapmotion.com/documentation/v4/concepts.html

[the remark this answers is not in the document]
Comment (rblenkinsopp) [reply Packages/Tracking/OpenXR/Runtime/Scripts/OpenXRLeapProvider.cs:324]: Well, that's embarrassing 🤦🏻 I had it in my head they were cm, should have double checked!

**Comment by MattGrayUL** [comment]
I have tested this and it works as I would expect, here's what I have tested:

logging "pinch strength" value

- tested OXR layer 1.4.3 with develop - result: fluctuating around 0.4 when fully pinched, varies a lot when rotating hand
- tested OXR layer 1.4.3 with this branch - result: value of 0 until half-pinched, then gradually up to 1, sticks to 1 when pinched, same at any rotation
- sanity test with LeapC with this branch - result: value of 0 until half-pinched, then gradually up to 1, sticks to 1 when pinched, same at any rotation

**Review by MattGrayUL — APPROVED** [review]

**Pull request merged**